client/state/stats/lists/selectors.js

```diff
@@ -59,7 +59,7 @@ export const getSiteStatsPostStreakData = createSelector(
 
 		if ( streakData && streakData.data ) {
 			Object.keys( streakData.data ).forEach( ( timestamp ) => {
-				const postDay = i18n.moment.unix( timestamp );
+				const postDay = i18n.moment.unix( timestamp ).locale( 'en' );
```

Comment:
Oh sorry, misread the PR description. We'll probably want to add comments for why we're hard coding the locale since these will stick out pretty badly in the code.